js/core/utils/shadow_dom.js

```diff
@@ -60,3 +60,33 @@ export function addShadowDomStyles($element) {
 
     root.adoptedStyleSheets = [constructedStyleSheet];
 }
+
+function getShadowDomElementsInViewPort(x, y, el) {
+    let result = [];
+
+    const rect = el.getBoundingClientRect();
+
+    if(rect) {
+        if(x >= rect.left && x <= rect.right - 1 && y <= rect.bottom - 1 && y >= rect.top) {
```

Comment:
Because `right` and `bottom` values are not in the rect. You can check it here: https://yari-demos.prod.mdn.mozit.cloud/en-US/docs/Web/API/Element/getBoundingClientRect/_sample_.basic.html
The `elementsFromPoint` will return incorrect element

Reply:
doesn't not strict comparison help?